### ISSUE TITLE
The purpose of calculating graph_balance in the code

### ISSUE THREAD

**Issue by fang-kun** [ouy7han/S2GSL#1]
The paper only talks about three losses, but I found graph_balance in the code as an additional item of loss as well, and I'm wondering what the purpose of this is

**Comment by fang-kun** [comment]
Also, why does the Dual_graph_learning class of the model.py file use different strategies for different datasets, for example:
` if self.data_name != 'laptop':
            sem_fusion_syn_graph_out = sem_fusion_syn_graph_out + bert_enc_outputs`
`        if self.data_name == 'twitter':
            representation = torch.cat(
                (a1_rate * (sem_fusion_syn_graph_out + bert_enc_outputs),
                 a2_rate * (syn_fusion_sem_graph_out + bert_enc_outputs),
                 a3_rate * (sem_concat_syn_graph_out + bert_enc_outputs)),
                dim=-1)`
I did not find a description related to either of these issues in the paper, and I'm curious to know the purpose of doing so

**Comment by ouy7han** [comment]
> The paper only talks about three losses, but I found graph_balance in the code as an additional item of loss as well, and I'm wondering what the purpose of this is

Thanks  for your question. The main purpose of graph_balance is to allow the two branches to learn useful information while also contributing to training stability. Due to space limitations in the paper, we did not provide a detailed explanation. If graph_balance is removed, the performance difference would not be significant, but the optimal hyperparameters would need to be re-tuned.

**Comment by ouy7han** [comment]
> Also, why does the Dual_graph_learning class of the model.py file use different strategies for different datasets, for example: ` if self.data_name != 'laptop': sem_fusion_syn_graph_out = sem_fusion_syn_graph_out + bert_enc_outputs` ` if self.data_name == 'twitter': representation = torch.cat( (a1_rate * (sem_fusion_syn_graph_out + bert_enc_outputs), a2_rate * (syn_fusion_sem_graph_out + bert_enc_outputs), a3_rate * (sem_concat_syn_graph_out + bert_enc_outputs)), dim=-1)` I did not find a description related to either of these issues in the paper, and I'm curious to know the purpose of doing so

Thanks for your insightful question. The representations output by BERT contain rich information from the original context, which may be helpful for the final sentiment judgment, while different datasets have varying needs for this representation information.

**Comment by ouy7han** [comment]
> Also, why does the Dual_graph_learning class of the model.py file use different strategies for different datasets, for example: ` if self.data_name != 'laptop': sem_fusion_syn_graph_out = sem_fusion_syn_graph_out + bert_enc_outputs` ` if self.data_name == 'twitter': representation = torch.cat( (a1_rate * (sem_fusion_syn_graph_out + bert_enc_outputs), a2_rate * (syn_fusion_sem_graph_out + bert_enc_outputs), a3_rate * (sem_concat_syn_graph_out + bert_enc_outputs)), dim=-1)` I did not find a description related to either of these issues in the paper, and I'm curious to know the purpose of doing so

Thanks for your insightful question. The representation output by BERT contain rich information from the original context, which may be helpful for the final sentiment judgment, while different datasets have varying needs for this representation information.

**Comment by fang-kun** [comment]
Thanks for your answer. But, "different datasets have varying needs for this representation information.", how was this conclusion reached and are there any definitive experimental results?

**Comment by ouy7han** [comment]
> Thanks for your answer. But, "different datasets have varying needs for this representation information.", how was this conclusion reached and are there any definitive experimental results?

Sorry, we haven't done detailed experiments in this regard. We mainly get some possible conclusions through various attempts. For the Twitter dataset, we found that adding bert_enc_output does perform better than not adding it. This may be because global contextual semantic information is more important for the Twitter dataset. Some sentences in the Twitter dataset do not have clear opinion information, so it is necessary to understand the whole contextual semantic information to make better judgments. In contrast, most of the sentences in the laptop dataset have relatively clear opinions, so structural information may be more important for laptops. Different model and methods may different. I hope this helps you. Thank you again for your question!
